crates/bevy_transform/src/hierarchy/hierarchy.rs

```diff
@@ -36,15 +41,32 @@ fn despawn_with_children_recursive_inner(world: &mut World, entity: Entity) {
     }
 }
 
+fn despawn_children(world: &mut World, entity: Entity) {
+    if let Some(mut children) = world.get_mut::<Children>(entity) {
+        for e in std::mem::take(&mut children.0) {
+            despawn_with_children_recursive_inner(world, e);
+        }
+    }
+}
+
 impl Command for DespawnRecursive {
     fn write(self, world: &mut World) {
         despawn_with_children_recursive(world, self.entity);
     }
 }
 
+impl Command for DespawnChildrenRecursive {
+    fn write(self, world: &mut World) {
+        despawn_children(world, self.entity);
+    }
+}
+
 pub trait DespawnRecursiveExt {
     /// Despawns the provided entity and its children.
     fn despawn_recursive(self);
+
+    /// Despawns the provided entity's children.
+    fn despawn_children(&mut self);
```

Comment:
While I like the terseness of this name, it goes against the pattern we've already established:

1. `despawn`: only despawns the entity
2. `despawn_recursive`: despawns the entity and its descendants

If we apply the "despawn" action to children, that should only remove the children, not their descendants. According to the current naming conventions, this should be something like `despawn_children_recursive`.

That being said, there are already some issues prior to this pr, such as the docs for `despawn_recursive` being wrong: "despawns the provide entity and its children" is not the actual behavior of `despawn_recursive`. It despawns the entity _and its descendants_. 

Happy to discuss the best way to resolve this situation. Some random thoughts:
* If we somehow make "despawn recursive" the default behavior for `despawn`, then we can use the terser name here. Recursive despawns are the default in godot, which sort of makes sense to me, especially given that "scenes" are spawned as node trees in a hierarchy in godot, and we do the same thing in bevy. I expect that generally people despawning an entity would want the descendants to go with it. I'm not actually convinced yet that this is a good idea for bevy, but its worth discussing. It would require adding implicit per-component behaviors to the normal despawn action (either directly or as a "reaction").
* We could rename `despawn_children` to `despawn_children_recursive`.
* We could accept the inconsistent naming and move on. Not a huge fan of this. I'd prefer to at least have a general idea of what this api should (ideally) look like.

Reply:
I like making recursive the default, it's probably what people want to use anyway

Reply:
While I think option 1 is the ideal option, it's definitely not actionable for this PR. As for between option two or three, I'd personally prefer to keep the PR as is, but any clear cut answer would be great. The fourth implied option would be to not add this functionality until we have the ideal api, but that's not very appealing either.

Reply:
Agreed. No sense in blocking a useful feature on a big overhaul. I would prefer consistency here in the short term. Can you change this to `despawn_children_recursive` and fix the documentation to properly describe the recursive behavior for both despawn_recursive and despawn_children_recursive?

Reply:
What about `despawn_descendants`?

Reply:
I dig it!

Reply:
It's too cryptic IMO.

Reply:
I agree, but it forces the user to contend with the realities of the situation, so I think I still prefer it. `despawn_children_recursive` also requires a bit of brain work to understand exactly what it does. `despawn_children` has the aforementioned consistency issues.  `despawn_descendants` does what it says on the tin and uses standard tree terminology. If we ever change the `despawn` semantics, we can move back to the less cryptic `despawn_children` naming.